### ISSUE TITLE
How to modify HTML/CSS output

### ISSUE THREAD

**Issue by ghost** [asciidoctor/asciidoctor#2273]
I would like to modify the default HTML5/CSS output by adding some custom markup and styling.  Based on googling around, I would guess that I have to copy the current HTML5 backend and make modifications from there, but I can't find any straightforward documentation on what to copy or where to put it.  How do I go about this?

FYI, I am using [Swagger2Markup](https://github.com/Swagger2Markup/swagger2markup), but hopefully this doesn't really affect the steps I need to take.

**Comment by mojavelinux** [comment]
The Asciidoctor processor is designed so that you can control 100% of the output that it generates. Obviously, we provide default output to provide something out of the box.

You can supply your own stylesheet using the stylesheet attribute. If you just want to add a few extra styles, then a docinfo file is more appropriate. See http://asciidoctor.org/docs/user-manual/#docinfo-file

There are two mechanism to create custom output. You can either supply a set of templates or you can write a custom converter. The benefit of the templates is that you only need to supply templates for the parts you want to change.

Please do not copy or modify the built-in HTML5 converter. That's an internal class and not meant to be extended or reused. (In other words, do it at your own risk).

The simplest way to go is to use the templates. Here's information on how to do that:

https://github.com/asciidoctor/asciidoctor.org/issues/80

In a nutshell, you can clone https://github.com/asciidoctor/asciidoctor-backends and copy the templates for the nodes you want to customize out of the slim/html5 folder (or you can just use the whole folder). You pass the directory of your custom templates using the template_dirs option.

**Comment by mojavelinux** [comment]
In Asciidoctor 2.0, overloading a built-in converter is easier than ever. Here's an example:

```rb
class MyHtml5Converter < (Asciidoctor::Converter.for 'html5')
  register_for 'html5'
  def convert_paragraph node
    %(<p>#{node.content}</p>)
  end
end
```

Just load that file into the Ruby runtime and Asciidoctor will use your customizations.

**Comment by mojavelinux** [comment]
The documentation now covers this topic pretty thoroughly. If there is something specific that is still confusing, please file a separate issue.
